src/models/Batch.php

```diff
@@ -97,6 +97,11 @@ public function destroy(): bool
 
     private function processEntities(array $idArr, AbstractConcreteEntity $model, FilterableColumn $column, Action $action, array $params): void
     {
+        // in case we update ownership, the param sent will contain only the target owner
+        if ($action === Action::Update) {
```

Comment:
I would still create an UpdateOwner action, and in here check if action == UpdateOwner, change the params and change the action to Update.